src/files-and-uploads/data/utils.js

```diff
@@ -1,11 +1,54 @@
 import { InsertDriveFile, Terminal, AudioFile } from '@edx/paragon/icons';
 import { ensureConfig, getConfig } from '@edx/frontend-platform';
+import JSZip from 'jszip';
+import saveAs from 'file-saver';
 import FILES_AND_UPLOAD_TYPE_FILTERS from './constant';
 
 ensureConfig([
   'STUDIO_BASE_URL',
 ], 'Course Apps API service');
 
+export const getDownloadZipFolder = async (selectedRows, courseId, onError) => {
```

Comment:
This is a fairly complex function that's just tested as part of the component. It would be great to have more detailed and separate unit tests for this function that can include feasible edge cases of what can go wrong.

Reply:
What are some edge cases that I should make sure to include in my tests?

Reply:
The tests you are added are very good, but integration tests that run this function in the context of the component itself. This requires a much more fine-grained unit test as well (I would call the function in a test file directly, not add all that overhead with react + testing-library). I saw that you already covered:
- succeeds if mockResponseData = { ok: true, blob: () => 'Data' }; and calls fetch multiple times (happy path);
- displays error message and does not break on: mockResponseData = { ok: false };

Now there is a good number of other cases for the function that should be tested. Tests I'd like to see:
- Promise rejected does not throw an error
- mockResponseData = { ok: true } and mockResponseData = { ok: true, blob: null } don't throw errors
- If you have one successful response and one unsuccessful, `saveAs` will be called exactly for only the asset that was successful
- selectedRows is not defined
- courseId is not defined or not valid
- If selected rows is an empty array, then expected behavior is called. I found a bug there: that will trigger the `else` condition, yet you call `selectedAssets[0].original`, which should result in an error. It's not clear what the purpose of the else condition is intended to be.
- selectedRows has members that don't have an '.original' accessor.

Reply:
Generally, it would be best to separate unit tests for every single function that lives in the api.js files, but that's completely out of scope for this PR, of course.